job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java

```diff
@@ -1329,4 +1341,19 @@ public void run() {
         }, timeout);
         return timer;
     }
+
+    /**
+     * Desc: Waits till the job is Completed
+     * @param jobId
+     * @throws ApiException
+     * @throws InterruptedException
+     */
+    private void waitUntilJobCompletes(String jobId) throws ApiException, InterruptedException {
```

Comment:
What happens if there is an issue with the job and it never completes, will this get stuck in the loop. Why not use a for loop to ensure you exit out after a certain number of tries.
```suggestion
    private void waitUntilJobCompletes(String jobId) throws ApiException, InterruptedException {
        for (int i = 0; i < 60; i++) {
            Job job = jobsApi.getJob(defaultPartitionId, jobId, jobCorrelationId);
            String currentJobStatus = job.getStatus();
            LOG.info("Job " + jobId + " current status: " + currentJobStatus);
            if (currentJobStatus.equalsIgnoreCase("completed")) {
                break;
            }
            else {
                Thread.sleep(500);
            }
        }
    }
```